services/motion/builtin/move_request.go

```diff
@@ -0,0 +1,217 @@
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/base/kinematicbase"
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// moveRequest is a structure that contains all the information necessary for to make a move call.
+type moveRequest struct {
+	config *motion.MotionConfiguration
+
+	planRequest        *motionplan.PlanRequest
+	kinematicBase      kinematicbase.KinematicBase
+	position, obstacle *replanner
+}
+
+// plan creates a plan using the currentInputs of the robot and the moveRequest's planRequest.
+func (mr *moveRequest) plan(ctx context.Context) (motionplan.Plan, error) {
+	inputs, err := mr.kinematicBase.CurrentInputs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: this is really hacky and we should figure out a better place to store this information
+	if len(mr.kinematicBase.Kinematics().DoF()) == 2 {
+		inputs = inputs[:2]
+	}
+	mr.planRequest.StartConfiguration = map[string][]referenceframe.Input{mr.kinematicBase.Kinematics().Name(): inputs}
+	return motionplan.PlanMotion(ctx, mr.planRequest)
+}
+
+func (mr *moveRequest) execute(ctx context.Context, plan motionplan.Plan) moveResponse {
+	waypoints, err := plan.GetFrameSteps(mr.kinematicBase.Kinematics().Name())
+	if err != nil {
+		return moveResponse{err: err}
+	}
+
+	// Iterate through the list of waypoints and issue a command to move to each
+	for i := 1; i < len(waypoints); i++ {
+		select {
+		case <-ctx.Done():
+			return moveResponse{}
+		default:
+			mr.planRequest.Logger.Info(waypoints[i])
+			if err := mr.kinematicBase.GoToInputs(ctx, waypoints[i]); err != nil {
+				// If there is an error on GoToInputs, stop the component if possible before returning the error
+				if stopErr := mr.kinematicBase.Stop(ctx, nil); stopErr != nil {
+					return moveResponse{err: errors.Wrap(err, stopErr.Error())}
+				}
+				// If the error was simply a cancellation of context return without erroring out
+				if errors.Is(err, context.Canceled) {
+					return moveResponse{}
```

Comment:
same thing here, I think this needs to `return moveResponse{err: ctx.Err()}` otherwise the return is nondeterministic when the context is cancelled